pkg/controllers/monitoring/monitoring-stack/conditions.go

```diff
@@ -0,0 +1,125 @@
+	AvailableMessage               = "Monitoring Stack is available"
+	ReconciledMessage              = "Monitoring Stack successfully reconciled"
+)
+
+func updateConditions(msConditions []v1alpha1.Condition, promConditions []monv1.PrometheusCondition, generation int64, recError error) []v1alpha1.Condition {
+	if len(msConditions) == 0 {
+		return []v1alpha1.Condition{
+			{
+				Type:               v1alpha1.AvailableCondition,
+				Status:             v1alpha1.ConditionUnknown,
+				Reason:             "None",
+				LastTransitionTime: metav1.Now(),
+			},
+			{
+				Type:               v1alpha1.ReconciledCondition,
+				Status:             v1alpha1.ConditionUnknown,
+				Reason:             "None",
+				LastTransitionTime: metav1.Now(),
+			},
+		}
+	}
+	var updatedConditions []v1alpha1.Condition
+
+	for _, mc := range msConditions {
+		switch mc.Type {
+		case v1alpha1.AvailableCondition:
+			available := updateAvailable(mc, promConditions, generation)
+			if !available.Equal(mc) {
+				available.LastTransitionTime = metav1.Now()
+			}
+			updatedConditions = append(updatedConditions, available)
+		case v1alpha1.ReconciledCondition:
+			reconciled := updateReconciled(mc, promConditions, generation, recError)
+			if !reconciled.Equal(mc) {
+				reconciled.LastTransitionTime = metav1.Now()
+			}
+			updatedConditions = append(updatedConditions, reconciled)
+		}
+	}
+
+	return updatedConditions
+}
+
+// updateAvailable gets existing "Available" condition and updates its parameters
+// based on the Prometheus "Available" condition
+func updateAvailable(ac v1alpha1.Condition, prometheusConditions []monv1.PrometheusCondition, generation int64) v1alpha1.Condition {
+	ac.ObservedGeneration = generation
+
+	prometheusAvailable, err := getPrometheusCondition(prometheusConditions, monv1.PrometheusAvailable)
+
+	if err != nil {
+		ac.Status = v1alpha1.ConditionUnknown
+		ac.Reason = PrometheusNotAvailable
+		ac.Message = CannotReadPrometheusConditions
+		return ac
+	}
+
+	if prometheusAvailable.Status != monv1.PrometheusConditionTrue {
+		ac.Status = v1alpha1.ConditionStatus(prometheusAvailable.Status)
+		ac.Reason = PrometheusNotAvailable
+		ac.Message = prometheusAvailable.Message
+		return ac
+	}
+	ac.Status = v1alpha1.ConditionTrue
+	ac.Reason = AvailableReason
+	ac.Message = AvailableMessage
+	return ac
+}
+
+// updateReconciled updates "Reconciled" conditions based on the provided error value and
+// Prometheus "Reconciled" condition
+func updateReconciled(rc v1alpha1.Condition, prometheusConditions []monv1.PrometheusCondition, generation int64, err error) v1alpha1.Condition {
+	rc.ObservedGeneration = generation
+
+	if err != nil {
+		rc.Status = v1alpha1.ConditionFalse
+		rc.Message = err.Error()
+		rc.Reason = "FailedToReconcile"
+		return rc
+	}
+	prometheusReconciled, err := getPrometheusCondition(prometheusConditions, monv1.PrometheusReconciled)
+
+	if err != nil {
+		rc.Status = v1alpha1.ConditionFalse
```

Comment:
in `updateAvailable()`, we set the status to unknown. Any reason to have it different here?

Reply:
Yes, good catch. Thanks

Reply:
Updated

Reply:
You resolved this by changing `updateAvailable` to set `ConditionFalse`. I was thinking `ConditionUnknown` makes more sense here.
Can you outline why you went with `ConditionFalse` instead?

Reply:
Ahh yeah right. `ConditionUnknown` makes probably more sense when there's an error reading the Prometheus conditions. I just noticed I missed some other things so I am going to update. Thanks